app/jinja_filters.py

```diff
@@ -190,31 +190,18 @@ def setAttributes(dictionary, attributes):
 
 
 @blueprint.app_template_filter()
-def question_as_legend(question):
-    question_type = question['type']
-
-    if question_type == 'MutuallyExclusive':
-        return True
-
+def should_wrap_with_fieldset(question):
     answers = question['answers']
-    more_than_one_question = len(answers) > 1
 
-    if more_than_one_question:
-        return True
-
-    answer = answers[0]
-    if 'type' in answer and any(
-        answer['type'] in answer_type
-        for answer_type in ['Checkbox', 'Radio', 'Date', 'Duration']
-    ):
+    if len(answers) > 1 and not any(answer['type'] == 'Date' for answer in answers):
```

Comment:
Do we still need the exception as per @JamesGardiner's earlier suggestion for when type is not present?

Reply:
I'd say not, `type` is mandatory and validated in schema validator.